test/service.spec.js

```diff
@@ -757,6 +790,38 @@
         return expect(didError).to.equal(true);
       });
     });
+    describe('Service date sync functions', function() { 
```

Comment:
Can you add a test confirming that two service clients can have different clock skews? Basically create two instances of MockService, send 2 mocked responses using `mockHttpResponse` where the `date` headers are different, and make sure each client's `systemClockOffset` is different and correct.